### PR TITLE
Feature/redbox 697 wikipedia tool without migrations

### PR DESCRIPTION
**Pull request by gecBurton**
## Changes proposed in this pull request

1. migrations have been removed
2. `original_resource_ref` -> `uri`
3. removed the need for reingest
4. bugs removed, code tidied

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [ ] I have run integration tests

### CODE REVIEW

File: django_app/redbox_app/redbox_core/consumers.py [hunk 1]
```diff
@@ -200,24 +200,25 @@ def save_ai_message(
         chat_message = ChatMessage(chat=session, text=user_message_text, role=ChatRoleEnum.ai, route=self.route)
         chat_message.save()
         for file, citations in self.citations:
-            file.last_referenced = timezone.now()
-            file.save()
+            if file:
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/consumers.py:203]: only update a file if it exists (this wont be the case for external citations)

File: django_app/redbox_app/redbox_core/consumers.py [hunk 2]
```diff
@@ -200,24 +200,25 @@ def save_ai_message(
...
                         page_numbers=parse_page_number(citation.metadata.get("page_number")),
-                        source=Citation.Origin.USER_UPLOADED_DOCUMENT
                     )
                 else:
                     Citation.objects.create(
                         chat_message=chat_message,
-                        url=citation.metadata.get("original_resource_ref"),
                         text=citation.page_content,
-                        source=Citation.Origin(citation.metadata.get("creator_type"))
+                        source=Citation.Origin.WIKIPEDIA,
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/consumers.py:220]: yes i know not all external citations will from wikipedia in the future, but this is the case in this PR

File: django_app/redbox_app/redbox_core/consumers.py [hunk 3]
```diff
@@ -265,30 +266,21 @@ async def handle_activity(self, response: dict):
...
+        for ref, sources in sources_by_resource_ref.items():
+            try:
+                file = await File.objects.aget(original_file=ref)
+                payload = {"url": str(file.url), "original_file_name": file.original_file_name}
+            except File.DoesNotExist:
+                file = None
+                payload = {"url": ref, "original_file_name": None}
+
+            await self.send_to_client("source", payload)
+            self.citations.append((file, sources))
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/consumers.py:283]: this is simpler

File: redbox-core/redbox/models/file.py [hunk 4]
```diff
@@ -25,8 +26,8 @@ class ChunkMetadata(BaseModel):
     index: int = 0  # The order of this chunk in the original resource
     created_datetime: datetime.datetime = datetime.datetime.now(datetime.UTC)
     chunk_resolution: ChunkResolution = ChunkResolution.normal
-    creator_type: str
-    original_resource_ref: str  # URL or file name
+    creator_type: Literal["wikipedia", "uploaded_file"]
+    uri: str = Field(validation_alias=AliasChoices("uri", "file_name"))  # URL or file name
```
Comment (gecBurton) [inline redbox-core/redbox/models/file.py:30]: this allows backwards compatibility, i.e. no need for a re-ingest

**Review by jamesrichards4 — APPROVED** [review]

**Pull request merged**